server/SaveLoad.cpp

```diff
@@ -540,7 +540,7 @@ void LoadEmpireSaveGameData(const std::string& filename, std::map<int, SaveGameE
     try {
         fs::path path = FilenameToPath(filename);
         DebugLogger() << "LoadEmpireSaveGameData: filename: " << filename << " path:" << path;
-        fs::ifstream ifs(path, std::ios_base::binary);
+        fs::ifstream ifs(path.generic_path(), std::ios_base::binary);
```

Comment:
Would need to bump minimum boost version to 1.63 (note this is commented as experimental).

path.generic [first appeared](https://github.com/boostorg/filesystem/commit/6e92c9a8b293c4b94969dd309405c80b67dacfdd) in boost version 1.60, with the rename to generic_path in 1.63

[make_preferred](https://www.boost.org/doc/libs/1_58_0/libs/filesystem/doc/reference.html#path-modifiers)?

Reply:
Maybe
````
#ifdef FREEORION_LINUX
        fs::ifstream ifs(path.native(), std::ios_base::binary);
#else
        fs::ifstream ifs(path, std::ios_base::binary);
#endif
````

Reply:
In this case path.native == path.string
If the path separator is suspected to be the issue, may want to convert to path.generic on client side before sending to server (most/all such functions on linux perform no conversion).
`IsExistingFile` could be used to validate at server.

Reply:
As #2205 has been reported for the release branch, a fix for it needs to be cherry picked. However, on the release branch we still work with v8 of the SDK, meaning, IIRC, the min boost version there is 1.58 or 1.59. I really don't want to have to upgrade the SDK/boost versions for the release branch, so a patch for #2205 must not rely on newer boost versions.